File: frontend/lib/models/advertisement.dart
class Advertisement {
  final String id;
  final String title;
  final String description;
  final String type;
  final String fileUrl;
  final String? fileName;
  final String platform;
  final String placement;
  final String position;
  final int fadeDurationSeconds;
  final bool isDismissible;
  final String? targetUrl;
  final String? businessName;
  final int displayOrder;
  final bool isActive;

  Advertisement({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.fileUrl,
    this.fileName,
    required this.platform,
    required this.placement,
    required this.position,
    required this.fadeDurationSeconds,
    required this.isDismissible,
    this.targetUrl,
    this.businessName,
    required this.displayOrder,
    required this.isActive,
  });

  factory Advertisement.fromJson(Map<String, dynamic> json) {
    return Advertisement(
      id: json['id']?.toString() ?? '',
      title: json['title']?.toString() ?? 'Advertisement',
      description: json['description']?.toString() ?? '',
      type: json['type']?.toString() ?? 'Image',
      fileUrl: json['fileUrl']?.toString() ?? '',
      fileName: json['fileName']?.toString(),
      platform: json['platform']?.toString() ?? 'Both',
      placement: json['placement']?.toString() ?? 'Popup',
      position: json['position']?.toString() ?? 'BottomRight',
      fadeDurationSeconds: json['fadeDurationSeconds'] is int
          ? json['fadeDurationSeconds']
          : int.tryParse(json['fadeDurationSeconds']?.toString() ?? '') ?? 8,
      isDismissible: json['isDismissible'] == true,
      targetUrl: json['targetUrl']?.toString(),
      businessName: json['businessName']?.toString(),
      displayOrder: json['displayOrder'] is int
          ? json['displayOrder']
          : int.tryParse(json['displayOrder']?.toString() ?? '') ?? 0,
      isActive: json['isActive'] == true,
    );
  }
}

File: backend/Models/Advertisement.cs
using System.ComponentModel.DataAnnotations;

namespace RightFitGigs.Models
{
    public class Advertisement
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();
        
        [Required]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;
        
        [StringLength(1000)]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        [StringLength(20)]
        public string Type { get; set; } = "Image"; // Image, Video
        
        [Required]
        [StringLength(500)]
        public string FileUrl { get; set; } = string.Empty;
        
        [StringLength(100)]
        public string? FileName { get; set; }
        
        [StringLength(50)]
        public string Platform { get; set; } = "Both"; // Mobile, Web, Both

        [StringLength(30)]
        public string Placement { get; set; } = "Popup"; // Popup, PinnedFade

        [StringLength(20)]
        public string Position { get; set; } = "BottomRight"; // TopLeft, TopRight, BottomLeft, BottomRight, Center

        public int FadeDurationSeconds { get; set; } = 8;

        public bool IsDismissible { get; set; } = true;
        
        [StringLength(500)]
        public string? TargetUrl { get; set; } // Click-through URL
        
        [StringLength(100)]
        public string? BusinessName { get; set; }
        
        public int DisplayOrder { get; set; } = 0;
        
        public bool IsActive { get; set; } = true;
        
        public DateTime StartDate { get; set; } = DateTime.UtcNow;
        
        public DateTime? EndDate { get; set; }
        
        public int ViewCount { get; set; } = 0;
        
        public int ClickCount { get; set; } = 0;
        
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedDate { get; set; } = DateTime.UtcNow;
        
        [StringLength(100)]
        public string CreatedBy { get; set; } = string.Empty;
    }
}
